src/AudacityApp.h

```diff
@@ -113,6 +114,8 @@ class AudacityApp final : public wxApp {
    std::unique_ptr<wxSocketServer> mIPCServ;
 #endif
 
+   std::unique_ptr<UpdateManager> mUpdateManager;
```

Comment:
Now I understand differently and I'm changing the suggestions of the previous review.

The key is to reason about lifetimes of objects.

The UpdateManager lives as long as the unique AudacityApp object.  (So perhaps it doesn't even need this unique_ptr.  It might just be a static variable, if there is no problem destroying it after the AudacityApp object.)  It has a timer and checks for updates periodically (twice a day).  That might repeat of Audacity is left open a long time.

But, project windows may be opened and closed in that long time, and Audacity might even be in a state where there are no windows (it can be done on macOS, which still leaves a menu bar at top of screen and allows you to switch in and out of the Audacity application.)

This has implications.

Reply:
@Paul-Licameli In case, when Audacity have no project windows (macOS case), I guess `FindProjectFrame( GetActiveProject() )` returned `nullptr` and I can't executing next actions below (May be I wrong).
```cpp
wxFrame* parent = FindProjectFrame(GetActiveProject());
parent->CallAfter([this, parent]{ /*some body*/ });
```

Reply:
Then write

`wxTheApp->CallAfter(...`